include/LightGBM/feature_group.h

```diff
@@ -110,14 +110,56 @@ class FeatureGroup {
   }
 
   /*!
-   * \brief Constructor from memory
+   * \brief Constructor from memory when data is present
    * \param memory Pointer of memory
    * \param num_all_data Number of global data
    * \param local_used_indices Local used indices, empty means using all data
+   * \param group_id Id of group
    */
-  FeatureGroup(const void* memory, data_size_t num_all_data,
+  FeatureGroup(const void* memory,
+               data_size_t num_all_data,
                const std::vector<data_size_t>& local_used_indices,
                int group_id) {
+    // Load the definition schema first
+    const char* memory_ptr = LoadDefinitionFromMemory(memory, group_id);
+
+    // Allocate memory for the data
+    data_size_t num_data = num_all_data;
+    if (!local_used_indices.empty()) {
+      num_data = static_cast<data_size_t>(local_used_indices.size());
+    }
+    AllocateBins(num_data);
+
+    // Now load the actual data
+    if (is_multi_val_) {
+      for (int i = 0; i < num_feature_; ++i) {
+        multi_bin_data_.back()->LoadFromMemory(memory_ptr, local_used_indices);
+        memory_ptr += multi_bin_data_.back()->SizesInByte();
```

Comment:
```suggestion
        multi_bin_data_[i]->LoadFromMemory(memory_ptr, local_used_indices);
        memory_ptr += multi_bin_data_[i]->SizesInByte();
```

Reply:
Otherwise, the data will always be loaded into the last bin.

Reply:
nice! great catch